circuit-codegen-annotations/build.gradle.kts

```diff
@@ -28,6 +28,7 @@ kotlin {
         dependencies {
           api(libs.anvil.annotations)
           api(libs.dagger)
+          api(libs.hilt)
```

Comment:
Let's actually remove all three of these dependencies now since we don't want to impose a specific framework anymore

Reply:
if the hilt meta-annotation is required for `CircuitInject`, would `compileOnly()` for the hilt dependency work?

I know for the JVM annotations don't actually need to be on the classpath at runtime

Reply:
That could work, though still not thrilled about requiring this at all. Could you or @jamiesanson explain it a bit more? I don't know that much about Hilt these days

Reply:
The `GeneratesRootInput` annotation exists for repeatability and correctness when using Hilt, due to how it detects modules. It allows the Hilt compiler to see that other processors are going to produce input for it, meaning it can wait until other codegen steps complete before it wires up its own dagger components. Hilt won't necessary mess things up if this annotation isn't there, but it's not reliable without it. Docs for hilt extensions are here if you're interested in a quick browse: https://dagger.dev/hilt/creating-extensions.html

It's definitely not needed at runtime, and is retained at class-level, so should be fine making this `compileOnly`